core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -1255,6 +1277,9 @@ private void writeValue(Object value) throws IOException {
             generator.writeEndObject();
         } else if (value instanceof Map) {
             writeMap((Map) value);
+        } else if (value instanceof Path) {
+            //Path implements Iterable<Path> and causes endless recursion and a StackOverFlow if treated as an Iterable here
+            generator.writeString(value.toString());            
```

Comment:
I'm not actually sure if `value.toString()` is equivalent or superior to `value.toAbsolutePath().toString()`.